### ISSUE TITLE
BinaryReadDocumentContext close throws and catches exception when !isPresent()

### ISSUE THREAD

**Issue by JerryShea** [OpenHFT/Chronicle-Wire#170]

**Comment by hft-team-city** [comment]
Released in [Chronicle-Wire-2.19.75](https://github.com/OpenHFT/Chronicle-Wire/releases/chronicle-wire-2.19.75), [BOM-2.19.359](https://github.com/OpenHFT/OpenHFT/releases/chronicle-bom-2.19.359)

**Comment by RobAustin** [comment]
see @peter-k-lawrey  comments below: 

```
The  MethodWriterInterceptors had many conflated concerns which were poorly tested/documented.

I would favour breaking out this functionality to a switch call you can make to turn output on and off.
I think this would make it's purpose clearer and still be easy to implement.

e.g.

interface MethodWriter {
    void marshallableOut(MarshallableOut out);
    // optionally
    void closeable(Closeable closeable);
}

During replay, it would be set to a /dev/null implementation.
```

**Comment by hft-team-city** [comment]
Released in [Chronicle-Wire-2.20.101](https://github.com/OpenHFT/Chronicle-Wire/releases/chronicle-wire-2.20.101), [BOM-2.20.134](https://github.com/OpenHFT/OpenHFT/releases/chronicle-bom-2.20.134)
